packages/core/src/common/path.ts

```diff
@@ -123,4 +123,20 @@ export class Path {
     toString(): string {
         return this.raw;
     }
+
+    isEqualOrParent(path: Path): boolean {
+        const value = this.toString();
+        const value2 = path.toString();
+        if (value === value2) {
+            return true;
+        }
+        if (!value || !value2) {
+            return false;
+        }
+        if (this.root && Path.isDrive(this.root.base)) {
+            return value2.startsWith(value);
+        }
+        return value2.toLowerCase().startsWith(value.toLowerCase());
```

Comment:
What about case-sensitive filesystems? Isn't there a corner case like `/home/a/foo` and `/home/a/Foo`?

Reply:
Oh ok so you test for case-sensitive systems first, and for non-case sensitive systems after?
Isn't `Path.isDrive(this.root.base)` meant to detect Windows though?

Reply:
you are right, it makes sense only in linux, I've not considered other since it is hard to decide on the frontend, I will assume that our fs api already returns normalized URIs.